### PR TITLE
[PAPI] Use "Timeout" for setting up a connectionTimeout

### PR DESCRIPTION
**Pull request by geropl**
## Related Issue(s)

Fixes #

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/dashboard/src/service/public-api.ts [hunk 1]
```diff
@@ -274,13 +275,18 @@ export function stream<Response>(
     const abort = new AbortController();
     (async () => {
         while (!abort.signal.aborted) {
+            const connectionTimeout = new Timeout(10_000);
             try {
+                connectionTimeout.start();
+
                 for await (const response of factory({
-                    signal: abort.signal,
+                    signal: AbortSignal.any([abort.signal, connectionTimeout.signal()!]),
```
Comment (geropl) [inline components/dashboard/src/service/public-api.ts:283]: TIL learned about `AbortSignal.any`, kind of made my day :laughing:

**Review by filiptronicek — APPROVED** [review]
LGTM, thanks!

**Comment by filiptronicek** [comment]
@geropl maybe we should get the `timeout.ts` from `gpl/672-sup-front` in here? Looks like there's some changes

**Comment by geropl** [comment]
> @geropl maybe we should get the timeout.ts from gpl/672-sup-front in here? Looks like there's some changes

Mostly tests and comments. I will make sure to merge them somewhere.

**Pull request merged**